Integrate Multipath QUIC discussion on Multicast channel (#133)
GrumpyOldTroll/draft-jholland-quic-multicast: draft-jholland-quic-multicast.md

A client with a matching joined channel always has at least one connection associated with the channel.
If a client has no matching joined channel, the packet is discarded.

## Channel using Multipath QUIC

> From the point of view of the client, each Multicast QUIC channel is handled as a second path from the server. A client keeps its unicast connection with the server open during all the transmission. Additionally, it negotiates a second path with the server where it will receive multicast content. All mechanisms, except those listed below, follow {{I-D.draft-ietf-quic-multipath}}.

Reviewer: ```suggestion
From the point of view of the client, each Multicast QUIC channel is handled as an additional path from the server. A client keeps its unicast connection with the server open during all the transmission. Additionally, the server can inform the client about an additional path where it will receive multicast content. All mechanisms, except those listed below, follow {{I-D.draft-ietf-quic-multipath}}.
```

Reviewer: This is mostly to make it clear that there can be more than one multicast channel and that it is server driven and not negotiated.

> The first major change concerns the encryption of the multicast path. To keep the unicast path between the client and the server secure, the multicast path must use a different cryptographic context that is common between all clients. As such, each path is encrypted differently, differing from the current version of {{I-D.draft-ietf-quic-multipath}}. NB: there are discussions within the Multipath QUIC draft to integrate this feature to strengthen the security of MPQUIC. Secondly, a client MUST NOT send any data on the multicast path with the server.

Reviewer: To me it's not quite clear if there is a need for a full cryptographic context at all for the channels or if the algorithm carried in announce frames plus the key from the key frames is enough. (Though I guess that is somewhat nitbitting)

Author: Indeed we have to specify what we mean by "a cryptographic context". As long as each data sent on the multicast channel is encrypted and decrypted with a different key than the unicast key (where I used the word "context"), it should be secure for the communication
